src/wasm-interpreter.h

```diff
@@ -2178,9 +2135,11 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     auto startVal = start.getSingleValue().getUnsigned();
     auto endVal = end.getSingleValue().getUnsigned();
     endVal = std::min<size_t>(endVal, refValues.size());
-    if (hasNonAsciiUpTo(refValues, endVal)) {
-      return Flow(NONCONSTANT_FLOW);
+
+    if (endVal > refValues.size()) {
```

Comment:
The previous line ensures that `endVal <= refValues.size()`, so this check is always false.